report/onionscanreport.go

```diff
@@ -65,8 +65,20 @@ type OnionScanReport struct {
 	SMTPFingerprint string `json:"smtpFingerprint"`
 	SMTPBanner      string `json:"smtpBanner"`
 
+	ProtocolInfoList []ProtocolInfo `json::"protocolInfoList"`
+
 	NextAction string `json:"lastAction"`
-	TimedOut   bool
+	TimedOut   bool   `json:"timedOut"`
+}
+
+type ProtocolInfo struct {
+	Type string      `json:"type"`
+	Port uint        `json:"port:`
+	Info interface{} `json:"info"`
+}
+
+func (osr *OnionScanReport) AddProtocolInfo(protocolType string, protocolPort uint, protocolInfo interface{}) {
+	osr.ProtocolInfoList = append(osr.ProtocolInfoList, ProtocolInfo{protocolType, protocolPort, protocolInfo})
```

Comment:
It's just the the BitcoinService was already so similar, and it seems like an elegant approach. But I'll hold it off for now.